rustls/src/lib.rs

```diff
@@ -248,6 +248,20 @@
 //!   which is used for cryptography.
 //!   Without this feature, these items must be provided externally to the core
 //!   rustls crate.
+//!
+//! - `aws_lc_rs`: this makes the rustls crate depend on the aws-lc-rs crate,
```

Comment:
I feel like this is where it gets tricky with the builder setup? As long as we have an implicit default it is possible for other dependencies to change the crypto provider without making it obvious, which seems like a bad idea.

Reply:
Did you have any thoughts on this?

Reply:
I think there are two classes of downstream users:

- would like TLS to work, but have little interest or opinion in detailed decisions like which `CryptoProvider` they are using. I think this sort of user can accept the default crate features and call `ClientConfig::builder()`. This has stable ("additive") behaviour irrespective of the choices made by other crates in their dependency tree.
- would like to configure details, and would call `ClientConfig::builder_with_provider()` to name a specific provider.

That leaves the possibility (with the PR as it currently stands) that someone is in the second camp, but (having opted out of this crate's default features and enabled `aws_lc_rs`) still call `ClientConfig::builder()`. This user would be a risk of getting a *ring*-backed `CryptoProvider` via their dependencies. Maybe we shouldn't provide aws-lc-rs-backed `builder()`?

Reply:
Yes, I think that would be better.

Reply:
I agree that I think it would be better as an explicit choice using `builder_with_provider`.